src/coreclr/jit/importer.cpp

```diff
@@ -7993,13 +7993,83 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 goto COND_JUMP;
 
             case CEE_SWITCH:
+#ifdef DEBUG
+                if (verbose)
+                {
+                    printf("\n---- CEE_SWITCH start in " FMT_BB " (PC=%03u) of '%s'", block->bbNum, block->bbCodeOffs,
+                           info.compFullName);
+                }
+#endif
                 /* Pop the switch value off the stack */
                 op1 = impPopStack().val;
                 assertImp(genActualTypeIsIntOrI(op1->TypeGet()));
 
-                /* We can create a switch node */
+                // Fold Switch for GT_CNS_INT
```

Comment:
You should try constant-folding the `op1` here, to increase the number of times we see constants.

```C++
                op1 = gtFoldExpr(impPopStack().val);
```

(replace line 8004 above)

Reply:
Done.